src/components/Context/createComponent.js

```diff
@@ -10,39 +10,47 @@ OF ANY KIND, either express or implied. See the License for the specific languag
 governing permissions and limitations under the License.
 */
 
+import { deepAssign, flatMap, isString, isFunction } from "../../utils";
+
 export default (config, logger, availableContexts, requiredContexts) => {
-  let configuredContexts;
-  return {
-    namespace: "Context",
-    lifecycle: {
-      onComponentsRegistered() {
-        let configuredContextNames = [];
+  let { context: configuredContexts } = config;
+
+  if (!Array.isArray(configuredContexts)) {
+    logger.warn("Invalid configured context. Please specify an array.");
+    configuredContexts = [];
+  }
 
-        if (Array.isArray(config.context)) {
-          configuredContextNames = config.context;
-        } else {
-          logger.warn(
-            `Invalid configured context. Please specify an array of strings.`
-          );
+  const contexts = flatMap(
+    requiredContexts.concat(configuredContexts),
```

Comment:
I considered that, but then I realized that they are able to overwrite the required items anyways by passing the data into the event payload.  Additionally, they may want to have the timestamp inside of their context function.

Reply:
> I considered that, but then I realized that they are able to overwrite the required items anyways by passing the data into the event payload.

Maybe we should move this code: https://github.com/adobe/alloy/blob/master/src/components/DataCollector/index.js#L67-L72 to before `onBeforeEvent` now that users can override automatically collected data using the functionality provided in this PR.

Reply:
Nevermind, that's not a good idea.

Reply:
Maybe?

Reply:
In what case would a customer want to override context data on a per-event basis that they couldn't do with a context function provided during configuration?

Reply:
We should bring this up with the larger group.

Reply:
Yeah, don't let it hold up the PR.

Reply:
In fact, if we started with the passed xdm and then applied the context functions, I think it would make them more useful.  The context functions could react to what was passed in.